app/DynamicsAdapter/DynamicsAdapter.Web/Services/Dynamics/DynamicService.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using DynamicsAdapter.Web.Configuration;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace DynamicsAdapter.Web.Services.Dynamics
+{
+    public class DynamicService : IDynamicService
+    {
+
+        private readonly AppSettings _settings;
+        private DateTime TokenExpiry;
+        private string AccessToken;
+       
+        public DynamicService(AppSettings settings)
+        {
+            _settings = settings;
+           
+        }
+        public async Task<JObject> Get(string entity)
+        {
+            RefreshToken();
+            using var client = new System.Net.Http.HttpClient
+            {
+                Timeout = new TimeSpan(0, _settings.DynamicsAPI.Timeout, 0)
+            };
+            client.DefaultRequestHeaders.Add("OData-MaxVersion", "4.0");
+            client.DefaultRequestHeaders.Add("OData-Version", "4.0");
+            client.DefaultRequestHeaders.Add("Prefer", "odata.include-annotations=\"*\"");
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", AccessToken);
+
+            var strResponse = await client.GetStringAsync(_settings.DynamicsAPI.EndPoints.Single(x => x.Entity == entity).URL);
+            return await  Task.FromResult(JObject.Parse(strResponse));
```

Comment:
Are we planning to use classes, not a generic JObject?

Reply:
There would be multiple entities, we should use generic JObject so this code can be reusable. We can use Template classes

Reply:
okay